tests/benchmarks/data_structure_perf/rbtree_perf/src/rbtree_perf.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <sys/rb.h>
+
+#define TREE_SIZE 512
+/* zephyr can't do floating-point arithmetic,
+ * so manual: dlog_N = 2log(TREE_SIZE) = 18
+ */
+const uint32_t dlog_N = 18;
+
+/* rbnode structure is embeddedable in user structure */
+struct container_node {
+	struct rbnode node;
+	int value;
+};
+static struct rbnode nodes[TREE_SIZE];
+static struct rbtree tree;
+
+/* Our "lessthan" is just the location of the struct */
+bool node_lessthan(struct rbnode *a, struct rbnode *b)
+{
+	return a < b;
+}
+
+/**
+ * @brief Test whether rbtree node struct is embedded
+ * in any user struct
+ *
+ * @details Initialize a user defined structure contains
+ * rbtree node.And into a rbtree and enumerate
+ * the rbtree nodes.
+ *
+ * verify that the value enumerated is equal to the
+ * value initialized.If the verification pass,the user
+ * defined structure works.
+ * verify "for each" style APIs work.
+ *
+ * @ingroup lib_rbtree_tests
+ *
+ * @see RB_FOR_EACH(),RB_FOR_EACH_CONTAINER()
+ */
+void test_rbtree_container(void)
+{
+	int count = 0;
+	struct rbtree test_tree_l;
+	struct container_node tree_node[10];
+	struct container_node *c_foreach_node;
+	struct rbnode *foreach_node;
+
+	test_tree_l.lessthan_fn = node_lessthan;
+	for (unsigned int i = 0; i < ARRAY_SIZE(tree_node); i++) {
+		tree_node[i].value = i;
+		rb_insert(&test_tree_l, &tree_node[i].node);
+	}
+
+	RB_FOR_EACH(&test_tree_l, foreach_node) {
+		zassert_true(CONTAINER_OF(foreach_node, struct container_node,
+		node)->value == count, "RB_FOR_EACH failed");
+		count++;
+	}
+
+	count = 0;
+
+	RB_FOR_EACH_CONTAINER(&test_tree_l, c_foreach_node, node) {
+		zassert_true(c_foreach_node->value == count,
+		"RB_FOR_EACH_CONTAINER failed");
+		count++;
+	}
+}
+
+/* initialize and insert a tree */
+void init_tree(struct rbtree *tree, int size)
+{
+	tree->lessthan_fn = node_lessthan;
+
+	for (int i = 0; i < size; i++) {
```

Comment:
previous function you use an unsigned int in the for loop, here you are using int, please make it consistent.